lang/rust-bootstrap/Portfile

```diff
@@ -77,109 +86,128 @@ depends_skip_archcheck      cmake \
                             ninja \
                             ccache
 
-dist_subdir                 rust
+if { ${configure.sdkroot} eq "" && ${os.platform} eq "darwin" && [vercmp ${macosx_deployment_target} > 10.8] } {
+    # allow universal build on macOS 10.13 with only 10.14 SDK, which is non-universal
+    # however, setting SDKROOT causes the build system to run `xcrun --show-sdk-path`, which older systems do not understand
+    # see https://github.com/rust-lang/cc-rs/blob/main/src/lib.rs
+    build.env-append        SDKROOT=/
+}
+
+# the downloaded Rust compiler must run on the machine
+# since binaries must run anyway, simplify build by using Rosetta and not worry about cross-compiling
+muniversal.run_binaries     yes
+triplet.add_host            all
+triplet.add_build           all
+foreach arch {arm64 x86_64 i386 ppc ppc64} {
+    muniversal.is_cross.${arch} [expr {${arch} eq ${configure.build_arch} ? "no" : "yes"}]
+}
 
 distname                    rustc-${version}-src
 distfiles                   ${distname}${extract.suffix}:apple_vendor
 
 # upstream source
 checksums                   rustc-${version}-src${extract.suffix} \
-                            rmd160  c6b0a06d789aacdc3bd4ab3f95d8a933273f8ba9 \
-                            sha256  b2bfae000b7a5040e4ec4bbc50a09f21548190cb7570b0ed77358368413bd27c \
-                            size    251552252
-
-set rustc_version           [rust.add_bootstrap_components ${configure.build_arch}]
-
-post-extract {
-    # ensure std library can be found by rust binary
-    foreach std_dir [glob -types d -directory ${workpath} -tails rust-std-*] {
-        set run_platform    [join [lrange [split ${std_dir} -] end-2 end] -]
-        set rustc_dir       rustc-[join [lrange [split ${std_dir} -] 2 end] -]
-        ln -s               ${workpath}/${std_dir}/rust-std-${run_platform}/lib/rustlib/${run_platform}/lib \
-                            ${workpath}/${rustc_dir}/rustc/lib/rustlib/${run_platform}/
-    }
+                            rmd160  c5cf1fa7214862c769fce8e2b582c9cdfa08c866 \
+                            sha256  882b584bc321c5dcfe77cdaa69f277906b936255ef7808fcd5c7492925cf1049 \
+                            size    283536892
+
+patchfiles-append           patch-offline_bootstrap.diff
+# see https://github.com/llvm/llvm-project/commit/a5e10e248efc72e7909e4067060e89c35a456a18
+patchfiles-append           patch-upstream_i386.diff
```

Comment:
Should this `patchfiles-append` be skipped on `arm64` builds?

Reply:
It would do no harm either way.
The patch file will eventually go away entirely as the changes have been accepted upstream.